diff --git a/lib/rucio/transfertool/fts3.py b/lib/rucio/transfertool/fts3.py
index 33dad3892..3cc0b2194 100644
--- a/lib/rucio/transfertool/fts3.py
+++ b/lib/rucio/transfertool/fts3.py
@@ -339,8 +339,8 @@ def build_job_params(
         # This functionality should reduce the number of stuck files in the disk buffer that are not migrated to tape media (for whatever reason).
         # Please be aware that FTS does not guarantee an atomic operation from the time it checks for existence of the file on disk and tape and 
         # the moment the file is overwritten, so there is a race condition that could overwrite the file on the tape media
-        overwrite = last_hop.rws.attributes.get('overwrite', False) # honour RSE configuration to force overwrite
-        overwrite_when_only_on_disk = last_hop.rws.attributes.get('overwrite_when_only_on_disk', False)
+        overwrite = last_hop.dst.rse.attributes.get('overwrite', False) # honour RSE configuration to force overwrite
+        overwrite_when_only_on_disk = last_hop.dst.rse.attributes.get('overwrite_when_only_on_disk', False)
         # setting both flags is incompatible, so we opt in for the safest approach: "overwrite_when_only_on_disk"
         # this is aligned with FTS implementation: see 
         if overwrite and overwrite_when_only_on_disk:
